### ISSUE TITLE
One line of script is said to have no issues detected but not runnable actually, hinted "find: paths must precede expression: `+exec'"

### ISSUE THREAD

**Issue by Ericwonne** [koalaman/shellcheck#2611]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

find . -iname "*.log" +exec rm -f;
```

#### Here's what shellcheck currently says:
```
No issues detected!

```
#### Here's what I wanted or expected to see:
I don't want to see what my terminal says: **find: paths must precede expression: `+exec'**, instead want it to work.
TODO: Describe expected/desired output

**Comment by Ericwonne** [comment]
maybe when it says "No issues detected!" doesn't mean it can run but only suggests that it has no syntax issue?

**Comment by brother** [comment]
The error is not in the shell syntax but in the order of flags for the `find` application.
Shellcheck have some knowledge about these flags, not sure I think it's the job of shellcheck to keep track of find flag order in this case.

**Comment by Ericwonne** [comment]
@brother I tried relocating the "-exec" part in the command for a few times, however all of which doesn't run properly. Would you please suggest some runnable version? thanks a lot :)

**Comment by brother** [comment]
@Ericwonne escape the semicolon (or use `+` to catch all files)
